### PR TITLE
Features/problem decoder

### PR DESCRIPTION
**Pull request by rickardenglund**

### CODE REVIEW

File: rest/problems.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/SKF/go-rest-utility/problems"
+)
+
+type ProblemDecoder interface {
```
Comment (Ekberg) [inline rest/problems.go]: Do we need to expose this interface in this package again?

File: rest/problems.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	DecodeProblem(context.Context, *http.Response) (problems.Problem, error)
+}
+
+type ComponentsProblemDecoder struct{}
```
Comment (Ekberg) [inline rest/problems.go]: I think we should leave this type as package private.

File: rest/problems_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/SKF/go-rest-utility/problems"
+	"github.com/stretchr/testify/require"
```
Comment (Ekberg) [inline rest/problems_test.go:13]: Should be moved up. Not sure if the linter is setup correctly in this repo

Comment (rickardenglund) [reply rest/problems_test.go:13]: why? those are 2 third party dependencies?

Comment (Ekberg) [reply rest/problems_test.go:13]: oh yeah, got the two PRs mixed up.

**Review by Ekberg — APPROVED** [review]

**Pull request merged**